src/shell/js/composeui-fdc3/src/ComposeUIDesktopAgent.ts

```diff
@@ -0,0 +1,312 @@
+import { ComposeUITopic } from './infrastructure/ComposeUITopic';
+
+//TODO sweep for non-standard errors
+export class ComposeUIDesktopAgent implements DesktopAgent {
+    private appChannels: ComposeUIChannel[] = [];
+    private userChannels: ComposeUIChannel[] = [];
+    private privateChannels: ComposeUIChannel[] = [];
+    private currentChannel?: ComposeUIChannel;
+    private messageRouterClient!: MessageRouter;
+    private currentChannelListeners: ComposeUIListener[] = [];
+
+    constructor(name: string, messageRouterClient: MessageRouter) {
+        this.messageRouterClient = messageRouterClient;
+        var channel = new ComposeUIChannel(
+            name,
+            "user",
+            this.messageRouterClient);
+        this.addChannel(channel);
+    }
+
+    //TODO
+    public open(app?: string | AppIdentifier, context?: Context): Promise<AppIdentifier> {
+        throw new Error("Not implemented");
+    }
+
+    //TODO
+    public findIntent(intent: string, context?: Context, resultType?: string): Promise<AppIntent> {
+        throw new Error("Not implemented");
+    }
+
+    //TODO
+    public findIntentsByContext(context: Context, resultType?: string): Promise<Array<AppIntent>> {
+        throw new Error("Not implemented");
+    }
+
+    //TODO
+    public findInstances(app: AppIdentifier): Promise<Array<AppIdentifier>> {
+        throw new Error("Not implemented");
+    }
+
+    public broadcast(context: Context): Promise<void> {
+        return new Promise((resolve, reject) => {
+            if (!this.currentChannel) {
+                reject(new Error("The current channel have not been set."));
+            } else {
+                resolve(this.currentChannel.broadcast(context));
+            }
+        });
+    }
+
+    //TODO
+    public raiseIntent(intent: string, context: Context, app?: string | AppIdentifier): Promise<IntentResolution> {
+        throw new Error("Not implemented");
+    }
+
+    //TODO
+    public raiseIntentForContext(context: Context, app?: string | AppIdentifier): Promise<IntentResolution> {
+        throw new Error("Not implemented");
+    }
+
+    //TODO
+    public addIntentListener(intent: string, handler: IntentHandler): Promise<Listener> {
+        throw new Error("Not implemented");
+    }
+
+    public addContextListener(contextType?: string | null | ContextHandler, handler?: ContextHandler): Promise<Listener> {
+        return new Promise<ComposeUIListener>(async (resolve, reject) => {
+            if (!this.currentChannel) {
+                reject(new Error("The current channel have not been set."));
+                return;
+            }
+            if (typeof contextType != 'string' || !contextType) {
+                reject(new Error("The contextType was type of ContextHandler, which would use a deprecated function, please use string or null for contextType!"));
+                return;
+            }
+
+            const stringContextType = contextType as string ?? null;
```

Comment:
Due to the previous condition, `contextType` is already narrowed to `string` here, no need for a new variable.